### PR TITLE
Sw 4 models

### PR DESCRIPTION
**Pull request by JedrzejMaluszczak**

### CODE REVIEW

File: backend/src/competitions/models/meet_models.py [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+    ('SOQ', 'swim-off after quarter-finals'),
+)
+
+
+class Pool(models.Model):
+    name = models.CharField(max_length=32, null=True)  # TODO: what's the max_length?
+    lane_min = models.IntegerField(null=True, validators=[
+        MinValueValidator(),  # TODO: what can be the min value?
+        MaxValueValidator(),  # TODO: what can be the max value?
+    ])
```
Comment (JedrzejMaluszczak) [inline backend/src/competitions/models/meet_models.py]:
```suggestion
   lane_min = models.IntegerField(
       null=True, 
       validators= [
           MinValueValidator(),  # TODO: what can be the min value?
           MaxValueValidator(),  # TODO: what can be the max value?
       ],
   )
```

File: backend/src/competitions/models/meet_models.py [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+    lane_max = models.IntegerField(null=True, validators=[
+        MinValueValidator(),  # TODO: what can be the min value?
+        MaxValueValidator(),  # TODO: what can be the max value?
+    ])
+    temperature = models.DecimalField(decimal_places=2, null=True, validators=[
+        MinValueValidator(Decimal(0)),
+        MaxValueValidator(Decimal(100)),
+    ])
```
Comment (JedrzejMaluszczak) [inline backend/src/competitions/models/meet_models.py]: ditto

File: backend/src/competitions/models/meet_models.py [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+    round = models.CharField(max_length=3, choices=ROUND, default='TIM')
+    gender = models.CharField(max_length=1, choices=GENDER_COMP, default='A')
+    max_entries = models.IntegerField(null=True, validators=[
+        MinValueValidator(),  # TODO: what's the min value?
+        MaxValueValidator(),  # TODO: what's the max value?
+    ])
+    run = models.IntegerField(default=1, validators=[
+        MinValueValidator(),  # TODO: what's the min value?
+        MaxValueValidator(),  # TODO: what's the max value?
+    ])
```
Comment (JedrzejMaluszczak) [inline backend/src/competitions/models/meet_models.py]: ditto

File: backend/src/competitions/models/results_models.py [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+    age_max = models.IntegerField(validators=[
+        MinValueValidator(),  # TODO: age_min
+        MaxValueValidator(99),
+    ])
+    gender = models.CharField(max_length=1, choices=GENDER_COMP)
+    name = models.CharField(max_length=64, null=True)  # TODO: what's the max_length?
+    number = models.IntegerField(null=True, validators=[
+        MinValueValidator(),  # TODO: what can be the min value?
+        MaxValueValidator(),  # TODO: what can be the max value?
+    ])
```
Comment (JedrzejMaluszczak) [inline backend/src/competitions/models/results_models.py]: ditto

File: backend/src/competitions/models/results_models.py [hunk 5]
```diff
@@ -0,0 +1,181 @@
...
+    day_time = models.DateTimeField(null=True)
+    final = models.CharField(max_length=1, choices=FINAL_TYPE, null=True)
+    number = models.IntegerField(validators=[
+        MinValueValidator(),  # TODO: what's the min value?
+        MaxValueValidator(),  # TODO: what's the max value?
+    ])
+    order = models.IntegerField(null=True, validators=[
+        MinValueValidator(),  # TODO: what's the min value?
+        MaxValueValidator(),  # TODO: what's the max value?
+    ])
```
Comment (JedrzejMaluszczak) [inline backend/src/competitions/models/results_models.py]: ditto

File: backend/src/competitions/models/results_models.py [hunk 6]
```diff
@@ -0,0 +1,181 @@
...
+class Result(models.Model):
+    status = models.CharField(max_length=3, choices=RESULT_STATUS, default='REG')
+    lane = models.IntegerField(null=True, validators=[
+        MinValueValidator(),  # TODO: what's the min value?
+        MaxValueValidator(),  # TODO: what's the max value?
+    ])
+    points = models.IntegerField(null=True, validators=[
+        MinValueValidator(),  # TODO: what's the min value?
+        MaxValueValidator(),  # TODO: what's the max value?
+    ])
```
Comment (JedrzejMaluszczak) [inline backend/src/competitions/models/results_models.py]: ditto

File: backend/src/competitions/models/results_models.py [hunk 7]
```diff
@@ -0,0 +1,181 @@
...
+    reaction_time = models.DecimalField(decimal_places=2, null=True, validators=[  # TODO: type (decimal?)?
+        MinValueValidator(),  # TODO: what's the min value?
+        MaxValueValidator(),  # TODO: what's the max value?
+    ])
```
Comment (JedrzejMaluszczak) [inline backend/src/competitions/models/results_models.py]: ditto

File: backend/src/competitions/models/results_models.py [hunk 8]
```diff
@@ -0,0 +1,181 @@
...
+    swim_time = models.DecimalField(decimal_places=2, null=True, validators=[  # TODO: type (decimal?)?
+        MinValueValidator(),  # TODO: what's the min value?
+        MaxValueValidator(),  # TODO: what's the max value?
+    ])
```
Comment (JedrzejMaluszczak) [inline backend/src/competitions/models/results_models.py]: ditto

File: backend/src/competitions/models/results_models.py [hunk 9]
```diff
@@ -0,0 +1,181 @@
...
+
+class Split(models.Model):
+    distance = models.IntegerField(validators=[
+        MinValueValidator(),  # TODO: what's the min value?
+        MaxValueValidator(),  # TODO: what's the max value?
+    ])
+    swim_time = models.DecimalField(decimal_places=2, validators=[  # TODO: type (decimal?)?
+        MinValueValidator(),  # TODO: what's the min value?
+        MaxValueValidator(),  # TODO: what's the max value?
+    ])
```
Comment (JedrzejMaluszczak) [inline backend/src/competitions/models/results_models.py]: ditto

File: backend/src/competitions/models/results_models.py [hunk 10]
```diff
@@ -0,0 +1,181 @@
...
+    lap_time = models.DecimalField(decimal_places=2, validators=[  # TODO: type (decimal?)?
+        MinValueValidator(),  # TODO: what's the min value?
+        MaxValueValidator(),  # TODO: what's the max value?
+    ])
```
Comment (JedrzejMaluszczak) [inline backend/src/competitions/models/results_models.py]: ditto

File: backend/src/competitions/models/shared_dictionaries.py [hunk 11]
```diff
@@ -0,0 +1,11 @@
+# TODO: do we want to list all possible nations?
+NATION = (
+
+)
+
```
Comment (JedrzejMaluszczak) [inline backend/src/competitions/models/shared_dictionaries.py]: ??

File: backend/src/competitions/models/meet_models.py [hunk 12]
```diff
@@ -0,0 +1,132 @@
+from django.db import models
+from django.core.validators import MaxValueValidator, MinValueValidator
+from decimal import Decimal
+
+from .shared_dictionaries import NATION, GENDER_COMP
+from .results_models import Competition
+
+POOL_TYPE = (
```
Comment (JedrzejMaluszczak) [inline backend/src/competitions/models/meet_models.py]: two blank lines after imports

File: backend/src/competitions/models/results_models.py [hunk 13]
```diff
@@ -0,0 +1,181 @@
+from django.db import models
+from django.core.validators import MaxValueValidator, MinValueValidator
+
+from .shared_dictionaries import GENDER_COMP
+from .meet_models import Event
+from .swimmer_models import Club
+
+STROKE = (
```
Comment (JedrzejMaluszczak) [inline backend/src/competitions/models/results_models.py]: two blank lines after imports

File: backend/src/competitions/models/swimmer_models.py [hunk 14]
```diff
@@ -0,0 +1,52 @@
+from django.db import models
+from django.core.validators import MaxValueValidator, MinValueValidator
+
+from .shared_dictionaries import NATION
+
+GENDER = (
```
Comment (JedrzejMaluszczak) [inline backend/src/competitions/models/swimmer_models.py]: two blank lines after imports

File: backend/src/competitions/models/swimmer_models.py [hunk 15]
```diff
@@ -0,0 +1,52 @@
...
+class Athlete(models.Model):
+    first_name = models.CharField(max_length=16)
+    last_name = models.CharField(max_length=16)  # TODO: what's the max length? (case: double last name)
+    license = models.CharField(max_length=32, null=True)  # TODO: what's the max_length?
+    birth_date = models.DateField()
+    gender = models.CharField(max_length=1, choices=GENDER)
+    nation = models.CharField(max_length=2, choices=NATION, null=True)  # TODO
+    swrid = models.IntegerField(null=True, validators=[  # TODO
+        MinValueValidator(0),
+    ])
```
Comment (JedrzejMaluszczak) [inline backend/src/competitions/models/swimmer_models.py]: ditto

File: backend/src/competitions/models/swimmer_models.py [hunk 16]
```diff
@@ -0,0 +1,52 @@
...
+class Club(models.Model):
+    short_name = models.CharField(max_length=20)
+    long_name = models.CharField(max_length=64, null=True)  # TODO: what's the max_length?
+    code = models.CharField(max_length=16, null=True)  # TODO: what_s the max length?
+    nation = models.CharField(max_length=2, choices=NATION, null=True)
+    number = models.IntegerField(null=True, validators=[  # TODO: `used to distinguish different teams of the same club
+        # TODO: in a meet entries or result file` (do we want it?)
+        MinValueValidator(),  # TODO: what can be the min value?
+        MaxValueValidator(),  # TODO: what can be the max value?
+    ])
```
Comment (JedrzejMaluszczak) [inline backend/src/competitions/models/swimmer_models.py]: ditto

File: backend/src/competitions/models/swimmer_models.py [hunk 17]
```diff
@@ -0,0 +1,52 @@
...
+    swrid = models.IntegerField(null=True, validators=[  # TODO
+        MinValueValidator(0),
+    ])
+
+
+class Club(models.Model):
+    short_name = models.CharField(max_length=20)
+    long_name = models.CharField(max_length=64, null=True)  # TODO: what's the max_length?
+    code = models.CharField(max_length=16, null=True)  # TODO: what_s the max length?
+    nation = models.CharField(max_length=2, choices=NATION, null=True)
```
Comment (JedrzejMaluszczak) [inline backend/src/competitions/models/swimmer_models.py]: reformat fields in all models to look like this:
```suggestion
    nation = models.CharField(
        max_length=2, 
        choices=NATION, 
        null=True,
    )
```

File: backend/src/urls.py [hunk 18]
```diff
@@ -5,7 +5,7 @@
 # Additionally, we include login URLs for the browsable API.
 from rest_framework import routers, views
 
-from competitions import api
+from .competitions import api
 
 router = routers.DefaultRouter()
```
Comment (JedrzejMaluszczak) [inline backend/src/urls.py]: two blank lines after imports

File: backend/src/competitions/models/swimmer_models.py (hunk 17 above)
Comment (MiloszKorman) [reply backend/src/competitions/models/swimmer_models.py]: do we want to write all fields in this convention, or only the ones with multiple parameters?

**Pull request closed without merging**